src/pages/dashboard/workplace/model.ts

```diff
@@ -0,0 +1,104 @@
+import { AnyAction, Reducer } from 'redux';
+import { EffectsCommandMap } from 'dva';
+import { ActivitiesType, CurrentUser, NoticeType, RadarDataType } from './data.d';
+import { fakeChartData, queryActivities, queryCurrent, queryProjectNotice } from './service';
+
+export interface ModalState {
+  currentUser: Partial<CurrentUser>;
+  projectNotice: NoticeType[];
+  activities: ActivitiesType[];
+  radarData: RadarDataType[];
+}
+
+export type Effect = (
+  action: AnyAction,
+  effects: EffectsCommandMap & { select: <T>(func: (state: ModalState) => T) => T },
+) => void;
+
+export interface ModelType {
+  namespace: string;
+  state: ModalState;
+  reducers: {
+    save: Reducer<ModalState>;
+    clear: Reducer<ModalState>;
+  };
+  effects: {
+    init: Effect;
+    fetchUserCurrent: Effect;
+    fetchProjectNotice: Effect;
+    fetchActivitiesList: Effect;
+    fetchChart: Effect;
+  };
+}
+
+const Model: ModelType = {
+  namespace: 'dashboardAndworkplace',
+  state: {
+    currentUser: {},
+    projectNotice: [],
+    activities: [],
+    radarData: [],
+  },
+  effects: {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [The key 'effects' is not sorted alphabetically](https://app.codacy.com/manual/afc163/ant-design-pro/pullRequest?prid=4583897)